File: leetcode/cpp/224.cpp
#include <bits/stdc++.h>

using namespace std;

int calculateWithoutBrace(vector<char> &s)
{
    int ret = 0;
    while (!s.empty())
    {
        int tempNum = 0;
        bool isPlus = true;
        bool isSign = false;
        if (s[0] == '-')
        {
            isPlus = false;
            isSign = true;
        }
        else if (s[0] == '+')
        {
            isSign =
        }
    }
    return ret;
}

int calculate(string s)
{
    vector<char> braceLeft, braceRight;

    for (int i = 0; i < s.size(); i++)
    {
        if (s[i] == ' ')
            continue;
        if (s[i] == ')')
        {
            while (braceLeft.back() != '(')
            {
                braceRight.insert(braceRight.begin(), braceLeft.back());
                braceLeft.pop_back();
            }
            string tempStr = to_string(calculateWithoutBrace(braceRight));
            for (int j = 0; j < tempStr.size(); j++)
            {
                braceLeft.push_back(tempStr[j]);
            }
            continue;
        }
        braceLeft.push_back(s[i]);
    }

    return calculateWithoutBrace(braceLeft);
}

int main()
{
    string s = "1 + 1";
    s = "(1+(4-8+2)-3)+(6+8)";
    cout << calculate(s) << endl;
    return 0;
}

File: leetcode/python/224.py
def calculate(s: str) -> int:
    s = s.replace(' ', '')
    i = 0
    while (i < len(s)):
        if s[i] == ')':
            j = i-1
            while (s[j] != '('):
                j -= 1
            s = s[:j]+str(calculate(s[j+1:i]))+s[i+1:]
            i = j
        else:
            i += 1
    if len(s) == 0:
        return 0
    if s[0] != '+' and s[0] != '-':
        s = '+'+s

    ret = 0
    i = 0

    while (i < len(s)):
        count_minus = 0
        is_plus = True
        while (s[i] == '+' or s[i] == '-'):
            if s[i] == '-':
                count_minus += 1
            i += 1
        is_plus = (count_minus % 2) == 0
        j = i
        while (j < len(s) and '0' <= s[j] <= '9'):
            j += 1
        ret += int(s[i:j]) if is_plus else -1*int(s[i:j])
        i = j

    return ret


# print(calculate("1 + 1"))
print(calculate("1+(     -2)"))
